activation/nipost.go

```diff
@@ -365,98 +396,109 @@ func (nb *NIPostBuilder) submitPoetChallenge(
 
 	logger.Debug("submitting challenge to poet proving service")
 
-	round, err := client.Submit(ctx, deadline, prefix, challenge, signature, nodeID)
+	registration := nipost.PoETRegistration{
+		NodeId:        nodeID,
+		ChallengeHash: types.Hash32(challenge),
+		Address:       client.Address(),
+	}
+
+	round, err := client.Submit(ctx, fetchProofDeadline, prefix, challenge, signature, nodeID)
 	if err != nil {
+		registration.RoundEnd = poetRoundEnd
+
+		if err := nipost.AddPoetRegistration(nb.localDB, registration); err != nil {
+			nb.logger.Warn("failed to save unsuccessful poet registration into db",
+				zap.Error(err),
+				log.ZShortStringer("smesherID", nodeID))
+		}
 		return nipost.PoETRegistration{},
 			&PoetSvcUnstableError{msg: "failed to submit challenge to poet service", source: err}
 	}
+
 	logger.Info("challenge submitted to poet proving service", zap.String("round", round.ID))
 
-	registration := nipost.PoETRegistration{
-		ChallengeHash: types.Hash32(challenge),
-		Address:       client.Address(),
-		RoundID:       round.ID,
-		RoundEnd:      round.End,
-	}
+	registration.RoundEnd = round.End
+	registration.RoundID = round.ID
 
-	if err := nipost.AddPoetRegistration(nb.localDB, nodeID, registration); err != nil {
+	if err = nipost.AddPoetRegistration(nb.localDB, registration); err != nil {
 		return nipost.PoETRegistration{}, err
 	}
-
-	return registration, err
+	return registration, nil
 }
 
 // submitPoetChallenges submit the challenge to registered PoETs
 // if some registrations are missing and PoET round didn't start.
 func (nb *NIPostBuilder) submitPoetChallenges(
 	ctx context.Context,
 	signer *signing.EdSigner,
-	poetProofDeadline time.Time,
-	curPoetRoundStartDeadline time.Time,
+	fetchProofDeadline,
+	curPoetRoundStartDeadline,
+	curPoetRoundEndDeadline time.Time,
 	challenge []byte,
 ) ([]nipost.PoETRegistration, error) {
-	// check if some registrations missing or were removed
 	nodeID := signer.NodeID()
 	registrations, err := nipost.PoetRegistrations(nb.localDB, nodeID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get poet registrations from db: %w", err)
 	}
 
-	registrationsMap := make(map[string]nipost.PoETRegistration)
+	successRegistrations := make(map[string]nipost.PoETRegistration)
+	validRegistrations := make(map[string]nipost.PoETRegistration)
+	missingRegistrations := []PoetService{}
```

Comment:
NIT: this is the only line in a block of multiple instantiations without a `make`

```suggestion
	missingRegistrations := make([]PoetService, 0)
```